Edit quiz

pachia-lee opened the pull request:
Edit quiz feature works now

pachia-lee committed:
Editing a fill blank question now works with db.

pachia-lee committed:
Multiple Choice questions can now be edited and updates db.

pachia-lee committed:
Can change quiz title in db now. Still need to change the title when it navigates back to the studio screen.

wangj97 committed:
multipleChoiceQuestion can choose more options, and History display Title->QuizId->startedTime

wangj97 committed:
Merge branch 'master' into newBranch

pachia-lee committed:
WIP on editing a quiz

pachia-lee committed:
WIP

pachia-lee committed:
Fill blannk questions now populate when editing a question

pachia-lee committed:
Title changes in studio when you change it in edit quiz

benderje26 commented on Models/BusinessLogic.cs:
```diff
@@ -80,29 +81,63 @@ public async Task SkipLogin() {
+    }
+    public void SetEditQuestionQuizManager(QuizManager quizManager) {
+        EditQuizManager = quizManager;
+    }
```
Could getEditQuizManager() and SetEditQuizManager be done with just the property

benderje26 commented on Models/IBusinessLogic.cs:
```diff
@@ -32,6 +32,9 @@ public interface IBusinessLogic : INotifyPropertyChanged {
     Task<(LogoutResult, string?)> Logout();
 
+    public QuizManager? getEditQuizManager();
+    void SetEditQuestionQuizManager(QuizManager quizManager);
```
Like earlier mentioned, we could put that property also in the IBusinessLogic which might simplify it and then just get rid of the methods

pachia-lee replied:
Thank you for your input! I've deleted the setter and getter method.

pachia-lee committed:
Merge branch 'master' into edit_quiz

benderje26 reviewed (dismissed):
Nice job! I appreciate you making this. If you could just consider what was mentioned that would be great

pachia-lee committed:
Deleted the getter method

pachia-lee committed:
Merge branch 'edit_quiz' of https://github.com/benderje26/QuizzingApp341 into edit_quiz

pachia-lee committed:
Deleted setter method

benderje26 committed:
Make mostly naming changes to the Create/Edit quiz objects

pachia-lee committed:
Added Delete Quiz Functionality.

Still need to make sure it works

pachia-lee committed:
Creating a new quiz now works! And deleting a quiz works too!

pachia-lee committed:
Deleting this screen because we are using EditQuiz instead.

pachia-lee committed:
Added functionality to update question numbers.

pachia-lee committed:
Added visibility functionality.

Still need to make sure it actually works.

pachia-lee committed:
Public and private quiz is working.

I added a column for public in quizzes table. And I changed the changevisibility method to make sure it changes correctly.

pachia-lee committed:
Deleted the All Quizzes title

pachia-lee committed:
Changed the title for editing questions

pachia-lee committed:
Added the DateCreated Property so that I can display it in the studio

pachia-lee committed:
Work in progress on activating a quiz.

pachia-lee committed:
Revert "Work in progress on activating a quiz."

This reverts commit d179ccb37270a78b11707af2ee504fa84bd068b4.

wangj97 committed:
Update QuizHistory.xaml

wangj97 committed:
Update QuizHistory.xaml

disable display Quiz ID

benderje26 committed:
Slightly simplified method, updated comments, and added support for both multiselect and single select

benderje26 committed:
Remove is from multiselect

benderje26 committed:
Merge branch 'newBranch' into edit_quiz

benderje26 committed:
Fixed and normalized some things for quiz editing

- Fixed fill blank question answers to property divide on commas
- Fixed ability in multiple choice to have varying amounts of answers
- Added ability for multiselect in creating multiple choice question
- Standardized formatting both visually in the app and in the code

benderje26 committed:
Remove unnecessary logic for multiselect vs single select :(

skogmanpe40 reviewed (approved):
Looks good
